sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/implementation/util/InternalContext.java

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.generic.core.implementation.util;
+
+import com.generic.core.util.Context;
+
+import java.util.Optional;
+
+/**
+ * Internal representation of {@link Context}.
+ */
+public interface InternalContext {
+    /**
+     * Get the key for the internal context.
+     * <p>
+     * For both {@code getKey} and {@code getValue} the value return is the last key-value pair added to the context.
+     *
+     * @return The key for the internal context.
+     */
+    Object getKey();
+
+    /**
+     * Get the value for the internal context.
+     * <p>
+     * For both {@code getKey} and {@code getValue} the value return is the last key-value pair added to the context.
+     *
+     * @return The value for the internal context.
+     */
+    Object getValue();
+
+    /**
+     * Get the number of key-value pairs in the internal context.
+     *
+     * @return The number of key-value pairs in the internal context.
+     */
+    int count();
```

Comment:
nit: should we call this `size` just to be consistent with the collection interfaces in Java?